packages/legend-extension-dsl-persistence/src/models/metamodels/pure/model/packageableElements/persistence/Persistence.ts

```diff
@@ -0,0 +1,575 @@
+      hashArray(this.connections),
+      this.targetShape,
+    ]);
+  }
+}
+
+/**********
+ * notifier
+ **********/
+
+export class Notifier implements Hashable {
+  notifyees: Notifyee[] = [];
+
+  get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.NOTIFIER,
+      hashArray(this.notifyees),
+    ]);
+  }
+}
+
+export abstract class Notifyee implements Hashable {
+  private readonly _$nominalTypeBrand!: 'Notifyee';
+
+  abstract get hashCode(): string;
+}
+
+export class EmailNotifyee extends Notifyee implements Hashable {
+  address!: string;
+
+  override get hashCode(): string {
+    return hashArray([PERSISTENCE_HASH_STRUCTURE.EMAIL_NOTIFYEE, this.address]);
+  }
+}
+
+export class PagerDutyNotifyee extends Notifyee implements Hashable {
+  url!: string;
+
+  override get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.PAGER_DUTY_NOTIFYEE,
+      this.url,
+    ]);
+  }
+}
+
+/**********
+ * target shape
+ **********/
+
+export abstract class TargetShape implements Hashable {
+  private readonly _$nominalTypeBrand!: 'TargetShape';
+
+  abstract get hashCode(): string;
+}
+
+export class MultiFlatTarget extends TargetShape implements Hashable {
+  modelClass!: PackageableElementReference<Class>;
+  transactionScope!: TransactionScope;
+  parts: PropertyAndFlatTarget[] = [];
+
+  override get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.MULTI_FLAT_TARGET,
+      this.modelClass.hashValue,
+      this.transactionScope,
+      hashArray(this.parts),
+    ]);
+  }
+}
+
+export class FlatTarget extends TargetShape implements Hashable {
+  modelClass!: PackageableElementReference<Class>;
+  targetName!: string;
+  partitionProperties: string[] = [];
+  deduplicationStrategy!: DeduplicationStrategy;
+  ingestMode!: IngestMode;
+
+  override get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.FLAT_TARGET,
+      this.modelClass.hashValue,
+      this.targetName,
+      hashArray(this.partitionProperties),
+      this.deduplicationStrategy,
+      this.ingestMode,
+    ]);
+  }
+}
+
+export class OpaqueTarget extends TargetShape implements Hashable {
+  targetName!: string;
+
+  override get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.OPAQUE_TARGET,
+      this.targetName,
+    ]);
+  }
+}
+
+export class PropertyAndFlatTarget implements Hashable {
+  property!: string;
+  flatTarget!: FlatTarget;
+
+  get hashCode(): string {
+    return hashArray([
+      PERSISTENCE_HASH_STRUCTURE.PROPERTY_AND_FLAT_TARGET,
+      this.property,
+      this.flatTarget,
+    ]);
+  }
+}
+
+export enum TransactionScope {
+  SINGLE_TARGET = 'SINGLE_TARGET',
+  ALL_TARGETS = 'ALL_TARGETS',
+}
+
+/**********
+ * deduplication strategy
+ **********/
+
+export abstract class DeduplicationStrategy implements Hashable {
+  private readonly _$nominalTypeBrand!: 'DeduplicationStrategy';
```

Comment:
Could we split this into smaller files and prefix the files with `DSLPersistence`? Thanks!

Reply:
Will do tmw and resolve.

Reply:
Ok, fixed in #990. Did the same with protocol.